console/console-init/ui/src/modules/iot-device/components/AddDeviceWithJson/AddDeviceWithJson.tsx

```diff
@@ -114,85 +123,145 @@ const AddDeviceWithJson: React.FunctionComponent<IAddDeviceWithJsonProps> = ({
       return false;
     }
   };
+
   const onFinish = () => {
     if (deviceDetail)
       if (!isJsonValid()) {
-        setShowJsonValidationError(true);
+        setErrorMessage("Invalid JSON syntax, unable to parse JSON");
       } else {
         onSave(deviceDetail);
       }
   };
+
   const handleOnPreview = () => {
     if (deviceDetail) {
       if (!isJsonValid()) {
-        setShowJsonValidationError(true);
+        setErrorMessage("Invalid JSON syntax, unable to parse JSON");
       } else {
-        onPreview(deviceDetail);
+        setIsPreviewEnabled(true);
       }
     }
   };
 
+  const handleOnBack = () => {
+    setIsPreviewEnabled(false);
+  };
+
+  const getDeviceDetail = () => {
+    const device: IDeviceProp = {
+      deviceInformation: {
+        metadata: []
+      },
+      credentials: [],
+      gateways: { gateways: [], gatewayGroups: [] }
+    };
+    if (deviceDetail) {
+      const obj = JSON.parse(deviceDetail);
```

Comment:
please use meaningful name instead of obj. like parseDeviceDetail etc. and  use destructuring